kolibri/auth/models.py

```diff
@@ -26,120 +38,258 @@ class KolibriValidationError(KolibriError):
     pass
 
 
-class BaseUser(AbstractBaseUser):
+##############################################
+# MORANGO: These pieces will live in Morango, but dummies/stubs are temporarily included here until Morango is ready.
+##############################################
```

Comment:
Looking through this block, I don't think anything in the rest of the code in the PR actually really depends on it. The role it played for me here was in exploring how best to decouple Kolibri and Morango (i.e. what pieces belonged on which side of the fence), as well as to think through some of the potential interactions between syncing and things like MPTT (with possible `tree_id` collisions).

It sounds like it would be cleanest if I removed this entire MORANGO block for now, and moved it to the notes/specs for Morango, so that it's clear that all of this also works without Morango (other than syncing, of course). The FacilityDataset stuff would remain, but that's not *part* of Morango -- it serves three purposes:
- Scoping data for use by permissions. For example, until a user has a learner role for a class/group, the FacilityDataset the main way we would determine who has permission to edit that user.
- Storing metadata about the facility -- since Collections only have a few key fields (like name, kind, and then the MPTT stuff), we need somewhere to store additional metadata.
- Filtering/partitioning for Morango, to make it easy/efficient to query for all model instances that are part of a specific set of "facility data". So, Morango will be told to *use* this field, but they're not tightly coupled.